item.rb

```diff
@@ -0,0 +1,46 @@
+class Item
+  attr_accessor :id, :publish_date, :archived
+  attr_reader :label, :genre, :author, :source
+
+  def initialize(id, publish_date, archived: false)
+    @id = id || Random.rand(1..100)
+    @publish_date = Date.parse(publish_date)
+    @archived = archived
+  end
+
+  def can_be_archived?
+    current_date = Date.today
+    year = current_date.year - @publish_date.year
+    year >= 10
+  end
+
+  def move_to_archive
+    @archived = true if can_be_archived?
+  end
+
+  def label=(label)
+    @label = label
+    label.items.push(self) unless label.items.include?(self)
+  end
+
+  def genre=(genre)
+    @genre = genre
+    genre.add_item(self) unless genre.items.include?(self)
+  end
+
+  def author=(author)
+    @author = author
+    author.items.push(self) unless author.items.include?(self)
+  end
+
+  def source=(source)
+    @source = source
+    source.items.push(self) unless source.items.include?(self)
```

Comment:
According to the UML diagram, we don't have reading permissions of the `items` field. We should use the `add_item` method, inside of which we should handle the case when the `items` field contains the item.